xlators/performance/io-cache/src/page.c

```diff
@@ -789,6 +789,8 @@ ioc_frame_unwind(call_frame_t *frame)
     GF_ASSERT(frame);
 
     local = frame->local;
+    /* gnfs requires op_errno to determine is_eof */
+    op_errno = local->op_errno;
```

Comment:
The following line checks if local is null, but you are accessing it right away prior to that. That doesn't look good - unless you know it isn't null somehow?

Reply:
I think you are right I could move this down. I'm glad someone is checking on this I don't trust myself here. I'm certainly not experienced with the code base or network programming.

At this time, I have the PR marked draft. I will try to make an isolated test case as suggested in the mailing list, be sure I can duplicate the problem (it wasn't a 100% fail), then I will test my patch including the fix you suggest above. I'll update the PR when I have that completed.